packages/client/components/ActivityLibrary/ActivityDetails/hooks/useActivityDetails.tsx

```diff
@@ -0,0 +1,141 @@
+        estimations to your backlog.
+      </>
+    ),
+    action: (
+      <>
+        This is a space to check in as a team. Share a personal update using the <b>Icebreaker</b>{' '}
+        phase. Give a brief update on what’s changed with your work during the <b>Solo Updates</b>{' '}
+        phase. Raise issues for discussion in the <b>Team Agenda</b> phase.
+      </>
+    )
+  },
+  integrationsTip: {
+    teamPrompt: <>push takeaway tasks to your backlog</>,
+    action: <>push takeaway tasks to your backlog</>,
+    retrospective: <>push takeaway tasks to your backlog</>,
+    poker: <>sync estimations with your backlog</>
+  }
+}
+
+type MeetingActivity = {
+  isTemplate: false
+}
+
+type TemplateActivity = {
+  isTemplate: true
+  template: ActivityDetailsQuery['response']['viewer']['availableTemplates']['edges'][number]['node']
+}
+
+export type Activity = {
+  id: ActivityId
+  name: string
+  illustration: string
+  usageStats?: any
+  category: CategoryID
+  categoryLink: string
+  type: MeetingTypeEnum
+  description: React.ReactNode
+} & (MeetingActivity | TemplateActivity)
+
+export const useActivityDetails = (
+  activityIdParam: string,
+  data: ActivityDetailsQuery['response']
+) => {
+  const {viewer} = data
+  const {availableTemplates} = viewer
+
+  const history = useHistory<{prevCategory?: string; edit?: boolean}>()
+  const prevCategory = history.location.state?.prevCategory
+
+  // for now, standups and check-ins don't have templates
+  const meetingIds = ['teamPrompt', 'action']
+  if (meetingIds.includes(activityIdParam)) {
+    const activityId = activityIdParam as ActivityId
+    const type = activityId as NoTemplatesMeeting
+    const category = MEETING_TYPE_TO_CATEGORY_ID[type]
+    const activity: Activity = {
+      id: activityId,
+      name: MEETING_ID_TO_NAME[type],
+      illustration: getActivityIllustration(activityId),
+      category,
+      categoryLink: `/activity-library/category/${
+        prevCategory ?? category ?? QUICK_START_CATEGORY_ID
+      }`,
+      type: activityId as MeetingTypeEnum,
+      description: ACTIVITY_TYPE_DATA_LOOKUP.description[type],
+      isTemplate: false
+    }
+
+    return {
+      activity
+    }
+  }
+
+  const selectedTemplate = availableTemplates.edges.find(
+    (edge) => edge.node.id === activityIdParam
+  )?.node
+  if (!selectedTemplate) return {activity: null}
+
+  const activityId = selectedTemplate.id as ActivityId
+  const category = selectedTemplate.category as CategoryID
+  const type = selectedTemplate.type
+
+  const activity: Activity = {
+    id: activityId,
+    name: selectedTemplate.name,
+    illustration: getActivityIllustration(activityId),
+    category,
+    categoryLink: `/activity-library/category/${
+      prevCategory ?? category ?? QUICK_START_CATEGORY_ID
+    }`,
+    type,
+    description: ACTIVITY_TYPE_DATA_LOOKUP.description[type],
+    isTemplate: true,
```

Comment:
+1 Let's get rid of this, since this is really just a convenience for `!!activity.template`

Reply:
Just like I said [here](https://github.com/ParabolInc/parabol/pull/8199#discussion_r1198874955), TS still complains about using `activity.template` in ternaries after doing `!!activity.template` so I'd rather keep the code more readable instead of using `?` and `!` in places where it's not actually needed.

Reply:
TIL that discriminating on the presence of an object isn't supported... weird! i'll bump ts & see if v5 supports it, I could've sworn it did...